### PR TITLE
49978_searchig_by_ransack

### PR DESCRIPTION
**Pull request by noor-framgia**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,7 +27,8 @@ gem "sdoc", "~> 0.4.0", group: :doc
 # gem "bcrypt", "~> 3.1.7"
 # Use Unicorn as the app server
 # gem "unicorn"
-
+gem "ransack"
+gem "will_paginate"
```
Comment (abdullahallmehedi) [inline Gemfile]: kaminari

File: app/controllers/admin/books_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 class Admin::BooksController < ApplicationController
-
   def index
+    @categories = Category.all
+    @search = Book.search(params[:q])
```
Comment (abdullahallmehedi) [inline app/controllers/admin/books_controller.rb]: ()

File: app/views/admin/books/index.html.erb [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
+      <%= f.label :title_cont, t(".title") %>
+      <%= f.text_field :title_cont %>
+
+      <%= f.label :category_id_eq, t(".category") %>
+      <%= f.select :category_id_eq,
+        options_from_collection_for_select(@categories,
+        :id, :name, @search.category_id_eq), {prompt: false},
+        {class: "form-control"} %>
+
+    <div class="actions"><%= f.submit  %></div>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: space

**Comment by Rownaknira** [comment]
ok

**Pull request merged**